task/common/identifier.go

```diff
@@ -12,38 +12,46 @@ import (
 	"github.com/aohorodnyk/uid"
 )
 
-type Identifier string
+type Identifier struct {
+	name string
+	salt string
+}
 
 var ErrWrongIdentifier = errors.New("wrong identifier")
 
 const (
 	maximumLongLength = 50
 	shortLength       = 16
```

Comment:
I find it curious that in most places in the code we don't refer to `shortLength`, but rather `shortLength/2`

Reply:
* Long identifiers are in the form `tpi-{1}-{2}-{3}` and are menat to be used for most cloud resources.
* Short identifiers are in the form `{2}{3}` and are meant to be used for a few cloud resources with severe name length constraints.

The `shortLength` constant specifies the total length of a short identifier, which is equal to the sum of the lengths for `{2}` and `{3}`. Thus, `{2}` and `{3}` are being generated with a legth limit of `shortLength/2`.

Reply:
* `{1}` is a normalized human-friendly name, generated from the `--name` parameter[^1] when provided, or randomly when omitted.
* `{2}` is either a random string (for random identifiers) or a deterministic string derived from `{1}` (for deterministic identifiers)
* `{3}` is just a hash of `{1}` and `{2}` so we can tell apart the cloud resources we created from any other resources


[^1]: e.g. if you run `task create --name=Example@Name` the resulting `{1}` will be `example-name`

Reply:
So `{3}` is just a checksum portion to make it less likely that somebody will create a resource name matching `tpi`'s conventions?

Reply:
Yes, `{3}` is just a small hash to tell apart any resource named with a `tpi-` (or `leo-` soon) prefix from resources we create.